lib/internal/DoublyLinkedList.js

```diff
@@ -60,3 +63,26 @@ DLL.prototype.shift = function() {
 DLL.prototype.pop = function() {
     return this.tail && this.removeLink(this.tail);
 };
+
+DLL.prototype.toArray = function () {
+    var arr = Array(this.length);
+    var idx = 0;
+    var curr = this.head;
+    for(idx = 0; idx < this.length; idx++) {
+        arr[idx] = curr.data;
+        curr = curr.next;
+    }
+    return arr;
+}
+
+DLL.prototype.filter = function (testFn) {
```

Comment:
I wanted to call it `filter` so that if someone wants to swap out the implementation of `_tasks` with an array, they could.  e.g.:

```js
var q = async.queue(worker);
q._tasks = [];
```

I remember we got an issue a while back after 2.0 because some people were doing custom splicing in the queue, which they couldn't do with the linked list.

It is strange than it's doesn't return a new list, though.